### DISCUSSION TITLE
glut/opengl2 linking error

### DISCUSSION THREAD

**Discussion by theJavin** [ocornut/imgui#6520] [New Users Build/Link/Run issues ONLY!]
I'm having linking errors while integrating Imgui into an existing project based on legacy OpenGL and glut. The project is a Cuda simulation that uses OpenGL for visualization and doesn't have make files at this time. Example output: 

/usr/bin/ld: /tmp/tmpxft_00004931_00000000-10_SVT.o: in function `drawPicture()':
tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x833d): undefined reference to `ImGui_ImplOpenGL2_NewFrame()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8342): undefined reference to `ImGui_ImplGLUT_NewFrame()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8347): undefined reference to `ImGui::NewFrame()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x834c): undefined reference to `ImGui::GetIO()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8367): undefined reference to `ImGui::ShowDemoWindow(bool*)'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x836c): undefined reference to `ImGui::Render()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8371): undefined reference to `ImGui::GetDrawData()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8379): undefined reference to `ImGui_ImplOpenGL2_RenderDrawData(ImDrawData*)'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8f76): undefined reference to `ImGui::GetDrawData()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0x8f7e): undefined reference to `ImGui_ImplOpenGL2_RenderDrawData(ImDrawData*)'
/usr/bin/ld: /tmp/tmpxft_00004931_00000000-10_SVT.o: in function `main':
tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb402): undefined reference to `ImGui::DebugCheckVersionAndDataLayout(char const*, unsigned long, unsigned long, unsigned long, unsigned long, unsigned long, unsigned long)'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb410): undefined reference to `ImGui::CreateContext(ImFontAtlas*)'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb415): undefined reference to `ImGui::GetIO()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb445): undefined reference to `ImGui::StyleColorsDark(ImGuiStyle*)'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb44a): undefined reference to `ImGui_ImplGLUT_Init()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb44f): undefined reference to `ImGui_ImplOpenGL2_Init()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb489): undefined reference to `ImGui_ImplOpenGL2_Shutdown()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb48e): undefined reference to `ImGui_ImplGLUT_Shutdown()'
/usr/bin/ld: tmpxft_00004931_00000000-5_SVT.cudafe1.cpp:(.text+0xb498): undefined reference to `ImGui::DestroyContext(ImGuiContext*)'
collect2: error: ld returned 1 exit status

I apologize if this is a really naive question but I'm new to Imgui and am working with an existing program I do research with. 

I'll greatly appreciate any help or advice.

**Comment by ocornut** [comment]
You are not linking against imgui source files and backend files.
